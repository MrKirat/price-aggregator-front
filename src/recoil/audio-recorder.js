import { atom } from 'recoil';

const isAudioRecorderEnabledState = atom({
    key: 'isAudioRecorderEnabledState',
    default: false
});

export default isAudioRecorderEnabledState;