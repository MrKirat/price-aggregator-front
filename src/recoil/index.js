export { default as isAudioRecorderEnabledState } from './audio-recorder';
export { default as isCameraEnabledState } from './camera';
export { default as searchStringState } from './search';
export { default as productListState } from './products';
export { default as isLoadingState } from './loading';