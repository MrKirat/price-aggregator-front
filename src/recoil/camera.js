import { atom } from 'recoil';

const isCameraEnabledState = atom({
    key: 'isCameraEnabledState',
    default: false
});

export default isCameraEnabledState;