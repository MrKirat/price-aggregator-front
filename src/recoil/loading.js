import { atom } from 'recoil';

const isLoadingState = atom({
    key: 'isLoadingState',
    default: false
});

export default isLoadingState;