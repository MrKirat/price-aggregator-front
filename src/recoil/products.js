import { atom } from 'recoil';

const productListState = atom({
    key: 'productListState',
    default: []
});

export default productListState;