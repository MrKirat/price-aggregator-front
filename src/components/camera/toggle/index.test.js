import React from 'react';

import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CameraToggle from './index';
import { RecoilRoot } from 'recoil';

it('renders without crashing', () => {
  const output = mount(<RecoilRoot><CameraToggle /></RecoilRoot>);
  expect(shallowToJson(output)).toMatchSnapshot();
});

it('renders with CameraToggle', () => {
  const output = mount(<RecoilRoot><CameraToggle /></RecoilRoot>);
  expect(output.find('CameraToggle')).toHaveLength(1);
});

it('renders with icon', () => {
  const output = mount(<RecoilRoot><CameraToggle /></RecoilRoot>);
  expect(output.find('svg')).toHaveLength(1);
});