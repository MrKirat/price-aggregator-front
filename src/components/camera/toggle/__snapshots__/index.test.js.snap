// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<RecoilRoot>
  <Batcher
    setNotifyBatcherOfChange={[Function]}
  />
  <CameraToggle>
    <ForwardRef
      className="makeStyles-icon-1"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(SvgIcon))
        className="makeStyles-icon-1"
        onClick={[Function]}
      >
        <ForwardRef(SvgIcon)
          className="makeStyles-icon-1"
          classes={
            Object {
              "colorAction": "MuiSvgIcon-colorAction",
              "colorDisabled": "MuiSvgIcon-colorDisabled",
              "colorError": "MuiSvgIcon-colorError",
              "colorPrimary": "MuiSvgIcon-colorPrimary",
              "colorSecondary": "MuiSvgIcon-colorSecondary",
              "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
              "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
              "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
              "root": "MuiSvgIcon-root",
            }
          }
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-1"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M14.12 4l1.83 2H20v12H4V6h4.05l1.83-2h4.24M15 2H9L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2zm-3 7c1.65 0 3 1.35 3 3s-1.35 3-3 3-3-1.35-3-3 1.35-3 3-3m0-2c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5z"
            />
          </svg>
        </ForwardRef(SvgIcon)>
      </WithStyles(ForwardRef(SvgIcon))>
    </ForwardRef>
  </CameraToggle>
</RecoilRoot>
`;
